src/lib/components/AccessRightField.js

```diff
@@ -111,13 +121,47 @@ export class AccessRightField extends Component {
   }
 }
 
-AccessRightField.propTypes = {
+class FormikAccessRightField extends Component {
+  render() {
+    // NOTE: This is a "cute" optimization.
+    //       In general, FastField only re-renders if
+    //       * formik slice associated with this.props.fieldPath changes
+    //         (i.e. `access` changes)
+    //       * props are ADDED or REMOVED to FastField
+    // So we add/remove a prop to FastField based on the presence of files.
+    // This way, FastField only renders when the things (access and hasFiles)
+    // it cares about change as it should be.
+    const change = this.props.hasFiles ? {change: true} : {}
```

Comment:
Comment: Out of curiosity, the `hasFiles` prop is coming from redux so I would guess that the `FormikAccessRightFIeld` component is being rerendered when the prop is changing and thus I would expect the ....Have you checked that the FastField is not rerendered without passing this custom prop? Actually by checking the code of [FastField](https://github.com/formium/formik/blob/master/packages/formik/src/FastField.tsx#L75) I don't see how that could work unless you pass the custom `shouldUpdate` prop and change the way your component is being updated...

Reply:
I had checked. It's the React shouldComponentUpdate rules in action: https://reactjs.org/docs/optimizing-performance.html#shouldcomponentupdate-in-action  . The gist of it is: when props/state changes, React always re-assesses if the component should be re-rendered, BUT tt doesn't always re-render the component; it's assessment might be: "no need to re-render". And FastField makes it so, unless for these reasons:  https://formik.org/docs/api/fastfield#overview